### PR TITLE
Make comparison of installed files case-insensitive

### PR DESCRIPTION
**Pull request by vicroms**
Fixes the check that prevents two ports from installing a file with the same name by making the comparison case insensitive.

### CODE REVIEW

File: src/vcpkg/commands.install.cpp [hunk 1]
```diff
@@ -234,11 +234,23 @@ namespace vcpkg
             // The VS2015 standard library requires comparison operators of T and U
             // to also support comparison of T and T, and of U and U, due to debug checks.
 #if _MSC_VER <= 1910
```
Comment (BillyONeal) [inline src/vcpkg/commands.install.cpp:236]: We don't care about VS2015 so you can delete this if you want. Not worth resetting testing if that's the only change.

File: azure-pipelines/e2e-ports/duplicate-file-a/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "duplicate-file-a",
+  "version": "0.0.1"
+}
```
Comment (BillyONeal) [inline azure-pipelines/e2e-ports/duplicate-file-a/vcpkg.json:4]: Please `format-manifest` these.

**Review by BillyONeal — APPROVED** [review]

**Comment by dg0yt** [comment]
This change leads to a CI regression in https://dev.azure.com/vcpkg/public/_build/results?buildId=107274&view=results:
- Android builds are on Linux with a case-sensitive filesystem.
- tmx installs `tmx.h`
- tmxparser installs `Tmx.h`.
- CI now reports 
 ~~~
  error: The following files are already installed in /mnt/vcpkg-ci/installed/arm64-android and are in conflict with tmxparser:arm64-android
  Installed by tmx:arm64-android  
  include/tmx.h

**Comment by dg0yt** [comment]
The CI run for the tool update, https://dev.azure.com/vcpkg/public/_build/results?buildId=107231&view=results, had a warning only:
~~~
warning: File /Users/vcpkg/Data/installed/arm64-osx/include/Tmx.h was already present and will be overwritten
~~~

**Comment by BillyONeal** [comment]
@dg0yt I don't believe this is a regression, I believe this was broken before and we are finally detecting that it is broken.

**Comment by dg0yt** [comment]
Last vcpkg.ci run now, https://dev.azure.com/vcpkg/public/_build/results?buildId=107290

x64-android, case-sensitive filesystem: No error.

x64-osx, case-insensitive filesystem: warning
~~~
warning: File /Users/vcpkg/Data/installed/arm64-osx/include/Tmx.h was already present and will be overwritten
~~~

But there is the error in that vcpkg.pr run. Only for two in three android triplets.!?

**Comment by dg0yt** [comment]
Hm, this PR here claims to "Make comparison of installed files case-insensitive" but actually changes "case_insensitive_ascii_less", not "case_insensitive_ascii_equals".

The particular case of interest has this input set (order undefined):
~~~
include/tmx.h
include/Tmx.h.in
include/Tmx.h
~~~

**Comment by dg0yt** [comment]
vcpkg.pr https://dev.azure.com/vcpkg/public/_build/results?buildId=107523 Errors:
~~~
influxdb-cxx:x64-osx:/include/Proxy.h
libproxy:x64-osx:/include/proxy.h

coin-or-cgl:x64-osx:/lib/libCgl.a
games101-cgl:x64-osx:/lib/libCGL.a
# (Debug variant differs by to 'd` suffix.)
~~~

All ports were built from source, due to the CMake update (?) and touching the baseline.

vcpkg.ci https://dev.azure.com/vcpkg/public/_build/results?buildId=107290 Only warnings about overwriting.

**Pull request merged**